Mathlib/Data/List/FinRange.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Kenny Lau, Scott Morrison
+
+! This file was ported from Lean 3 source module data.list.fin_range
+! leanprover-community/mathlib commit 9003f28797c0664a49e4179487267c494477d853
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.List.OfFn
+import Mathlib.Data.List.Perm
+
+/-!
+# Lists of elements of `Fin n`
+
+This file develops some results on `finRange n`.
+-/
+
+
+universe u
+
+namespace List
+
+variable {α : Type u}
+
+@[simp]
+theorem map_coe_finRange (n : ℕ) : ((finRange n) : List (Fin n)).map (Fin.val) = List.range n := by
+  simp_rw [finRange, map_pmap, Fin.val_mk, pmap_eq_map]
+  exact List.map_id _
+#align list.map_coe_fin_range List.map_coe_finRange
+
+theorem finRange_succ_eq_map (n : ℕ) : finRange n.succ = 0 :: (finRange n).map Fin.succ := by
+  apply map_injective_iff.mpr Fin.val_injective
+  rw [map_cons, map_coe_finRange, range_succ_eq_map, Fin.val_zero, ← map_coe_finRange, map_map,
+    map_map]
+  simp only [Function.comp, Fin.val_succ]
+#align list.fin_range_succ_eq_map List.finRange_succ_eq_map
+
+-- Porting note TODO: we have List.finRange_map_get in Data.List.Range, which seems to be exactly
+-- this but currently not a simp lemma. Should we remove map_nth_le, make finRange_map_get
+-- a simp lemma and align map_nth_le with finRange_map_get?
+/-@[simp]
+theorem map_nth_le (l : List α) : ((finRange l.length).map fun n => l.nthLe n n.2) = l :=
+  (ext_nthLe (by rw [length_map, length_finRange])) fun n _ h => by
+    rw [← nthLe_map_rev]
+    congr
+    · rw [nthLe_finRange]
+      rfl
+    · rw [length_finRange]
+      exact h
+#align list.map_nth_le List.map_nth_le
+-/
+
+theorem of_fn_eq_pmap {α n} {f : Fin n → α} :
```

Comment:
Also elsewhere
```suggestion
theorem ofFn_eq_pmap {α n} {f : Fin n → α} :
```

Reply:
done in all places, and also `of_injective` to `ofInjective`